platforms/core-execution/build-cache-local/src/main/java/org/gradle/caching/local/internal/DirectoryBuildCacheServiceFactory.java

```diff
@@ -81,11 +85,16 @@ public BuildCacheService createBuildCacheService(DirectoryBuildCache configurati
         checkDirectory(target);
 
         int removeUnusedEntriesAfterDays = configuration.getRemoveUnusedEntriesAfterDays();
-        Supplier<Long> removeUnusedEntriesOlderThan = TimestampSuppliers.daysAgo(removeUnusedEntriesAfterDays);
+
         describer.type(DIRECTORY_BUILD_CACHE_TYPE).
             config("location", target.getAbsolutePath()).
             config("removeUnusedEntriesAfter", removeUnusedEntriesAfterDays + " days");
 
+        // Use the days configured for the local build cache, or the global 'createdResources' config if not specified.
+        Supplier<Long> removeUnusedEntriesOlderThan = removeUnusedEntriesAfterDays < 1
```

Comment:
🤔 I see that when we deprecated `org.gradle.cache.cleanup` [user home definition overrides project definiton](https://github.com/gradle/gradle/blob/master/platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc#deprecated-orggradlecachecleanup-property). But here it's the other way around.

I wonder if we should the same strategy here?

Reply:
Oh, that property was under user home.

Reply:
Unfortunately, to make the new `buildCache.setRemoveUnusedEntriesAfterDays` take precedence over the (now deprecated) `DirectoryBuildCache` method would require significant complexity in the former.
With this PR, if a user calls `DirectoryBuildCache.setRemoveUnusedEntriesAfterDays()` with a non-default value, then they will get a deprecation warning but the value will be honoured. If they don't call that deprecated method, then the value will be obtained via the new API.